src/generators/non_gradient_based/feature_tweak/feature_tweak.jl

```diff
@@ -39,6 +39,21 @@ function feature_tweaking(
     x::AbstractArray,
     target::RawTargetType,
 )
+    if !(M isa Models.TreeModel)
+        throw(
+            ArgumentError(
+                "The feature tweak generator is only available for tree-based models."
+            ),
+        )
+    end
+    if M.likelihood == :classification_multi
+        throw(
+            ArgumentError(
+                "The feature tweak generator is only available for binary classification problems.",
+            ),
+        )
+    end
```

Comment:
can this be formatted on fewer lines?

Reply:
I'll move this over into the refactoring generators branch in whole and try to change the formatting there, it's making tests fail here and I don't want to fix this problem in this branch